### PR TITLE
[FLINK-35984][transform] Fix: Job crashes when metadata column names present in transform rules

### PR DESCRIPTION
**Pull request by MOBIN-F**
since metadata column existence check was done by searching identifier string in statement, without considering any syntax info

In the transform rule (projection: \*, '_namespace_nameschema_nametable_name_' AS string_literal) mentioned in FLINK-35984, Caused by: java.lang.IllegalArgumentException: wrong number of arguments is thrown because of the "\*" symbol. The PreTransformOperator introduced in FLINK-35272 fixes the "\*" problem, but there are still some scenarios where Caused by: java.lang.IllegalArgumentException: wrong number of arguments is thrown due to metaColumn name conflicts. This PR will fix these bugs.

### CODE REVIEW

**Comment by MOBIN-F** [comment]
cc @yuxiqian

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/parser/TransformParser.java [hunk 1]
```diff
@@ -498,29 +498,15 @@ private static SqlSelect parseProjectionExpression(String projection) {
         return parseSelect(statement.toString());
     }
 
-    private static List<Column> copyFillMetadataColumn(
-            String transformStatement, List<Column> columns) {
+    private static List<Column> copyFillMetadataColumn(List<Column> columns) {
+        // Add meteColumn for SQLValidater.validate
```
Comment (leonardBang) [inline flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/parser/TransformParser.java:502]: metaColumn

File: flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/UnifiedTransformOperatorTest.java [hunk 2]
```diff
@@ -917,6 +917,52 @@ public void testMetadataAndCalculatedTransform() throws Exception {
                 .destroyHarness();
     }
 
+    @Test
+    public void testMetadataTransformIncludeMetaColumnString() throws Exception {
+        TableId tableId = TableId.tableId("my_company", "my_branch", "schema_nullability");
+        UnifiedTransformTestCase.of(
+                        tableId,
+                        "id, name, age, id + age as computed, __table_name__ as metaCol1, '__table_name__' as metaCol2, '__namespace__name__schema__name__table__name__' as metaCol3, UPPER(__schema_name__) as metaCol4",
```
Comment (leonardBang) [inline flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/UnifiedTransformOperatorTest.java:925]: could you also add `__namespace_name__ as metaColNameSpaceName`, `__schema_name__ as metaColSchemaName` in this test?

**Review by leonardBang — COMMENTED** [review]
Thanks @MOBIN-F for the contribution, I only left two minor comments

File: flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/UnifiedTransformOperatorTest.java (hunk 2 above)
Comment (MOBIN-F) [reply flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/transform/UnifiedTransformOperatorTest.java]: done

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/parser/TransformParser.java (hunk 1 above)
Comment (MOBIN-F) [reply flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/parser/TransformParser.java]: done

**Review by yuxiqian — APPROVED** [review]
Looks great!

**Review by leonardBang — APPROVED** [review]
+1, wait the CI green

**Pull request merged**